10 LoaderSpinner/readme.md

```diff
@@ -0,0 +1,294 @@
+```
+
+## Create a Spinner component
+
+First, lets create a new folder to hold the component and its relative files.
+### ./src/common/spinner/loadingSpinner.tsx
+
+This componnet will display the spinner.
+
+```tsx
+import * as React from 'react';
+import './loadingSpinner.css';
+
+import PropTypes from 'prop-types';
+
+
+const InnerLoadingSpinerComponent:React.StatelessComponent<PropTypes> = (props) => {
+   if (props.trackedPromiseInProgress === true) {
+    return (
+      <div className="loading">
+        <div className="loading__background">
+          <div className="loading__backdrop">
+            <div className="loading__box">
+              <div className="loading__icon"></div>
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  } else { return null } 
+}
+
+ InnerLoadingSpinerComponent.propTypes = {
+  trackedPromiseInProgress: PropTypes.bool.isRequired,
+} 
+
+
+```
+
+We will also add a CSS file to povide some style to our spinner:
+
+### ./src/common/spinner/loadingSpinner.css
+```css
+.loading__background {
+	position: fixed;
+	width: 100%;
+	top: 0px;
+	bottom: 0px;
+	z-index: 1070;
+}
+
+.loading__backdrop {
+	height: 100%;
+	background-color: #000;
+	opacity: 0.5;
+}
+
+.loading__box {
+	height: 100%;
+	display: flex;
+	justify-content: center;
+	align-items: center;
+}
+
+.loading__icon {
+	position: relative;
+	width: 200px;
+	height: 200px;
+	border: 20px solid #868e96;
+	border-top-color: #007bff;
+	border-radius: 50%;
+	animation: spin .6s linear infinite;
+}
+
+@keyframes spin {
+	to {
+		transform: rotate(360deg);
+	}
+}
+
+```
+
+Now that we have created our component, we will need to use the HOC provided by [react-promise-tracker](https://github.com/Lemoncode/react-promise-tracker) to wrap the component.
+### ./src/common/spinner/loadingSpinner.tsx
+
+```diff
+import * as React from 'react';
+import './loadingSpinner.css';
++ import { promiseTrackerHoc } from 'react-promise-tracker';
+import PropTypes from 'prop-types';
+
+
+const InnerLoadingSpinerComponent:React.StatelessComponent<PropTypes> = (props) => {
+   if (props.trackedPromiseInProgress === true) {
+    return (
+      <div className="loading">
+        <div className="loading__background">
+          <div className="loading__backdrop">
+            <div className="loading__box">
+              <div className="loading__icon"></div>
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  } else { return null } 
+}
+
+ InnerLoadingSpinerComponent.propTypes = {
+  trackedPromiseInProgress: PropTypes.bool.isRequired,
+} 
++ export const LoadingSpinnerComponent = promiseTrackerHoc(InnerLoadingSpinerComponent);
+
+```
+
+We will also need to create a barrel to ease the access to the spinner component.
+
+### ./src/common/spinner/index.ts
+
+
+```ts
+export {LoadingSpinnerComponent} from './loadingSpinner';
+```
+
+After this, we only need to use our tracker whenever we use a promise.
```

Comment:
After this, we only need to wrap our asynchronous call with a _trackpromise_ statement and we will have all the magic done.